### PR TITLE
ISPN-9016 Remove .gitignore pattern for *.dat files

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-9016

IDEs automatically copy resources from the source directories to the
target directory during compilation, so old component metadata files
left over in the sources directory can overwrite the new component
metadata written by the Maven build directly in the target directory.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -30,19 +30,11 @@ classes
 *.swp
 # buffered files generated by editors ending in ~
 *~
-# testsuite generated files/folders
-PutObjectStoreDirHere
```
Comment (anistor) [inline .gitignore]: My local env used to be full of PutObjectStoreDirHere and ObjectStore and index leftovers from query, like /query/person. Are they gone or why are we no longer gitignore them?

Comment (danberindei) [reply .gitignore]: They should be gone, but maybe not from all modules... if you notice new `PutObjectStoreDirHere` directories, please create an issue in JIRA

**Comment by anistor** [comment]
I just came across PutObjectStoreDirHere and ObjectStore folders again in client/hot-rod. It's probable they are only created when running tests from Intellij. Still, that's enough reason to keep that garbage on the ignore list.

**Comment by danberindei** [comment]
Ok, let's say I accept `PutObjectStoreDirHere`, because that's the default in JBossTS. But `ObjectStore` is not the default, and I haven't found any reference to it in our code base, so I'd really like to remove it :)

**Comment by anistor** [comment]
why? it's garbage

**Comment by danberindei** [comment]
Because it means a test is not just using the default JBossTS by mistake, but it's going out of it's way to use a custom JBossTS configuration. 
`commons-test/src/main/resources/test-jbossts-properties.xml` already defines a volatile store, so if a test is able to use a custom JBossTS configuration it should be using this one (and most tests are already using it).

**Comment by anistor** [comment]
There-s still one unused/useless .gitignore left in infinispan-server-cli

**Comment by anistor** [comment]
Removed the remaining one in infinispan-server-cli and integrated. Thanks @danberindei !

**Comment by anistor** [comment]
I really regret merging this. After switching to topic branch of jdg-7.2.x and running some tests I have a big pile of junk I should normally not have to worry about. Can we backport the ObjectStore and component-metadata.dat fixes to jdg-7.2.x so it behaves identically to master? 
```
adrian@mojo ~/work/infinispan $ git status 
On branch jdg-7.2.x
Untracked files:
  (use "git add <file>..." to include in what will be committed)

	cli/cli-interpreter/src/main/resources/infinispan-cli-interpreter-component-metadata.dat
	client/hotrod-client/ObjectStore/
	core/src/main/resources/infinispan-core-component-metadata.dat
	core/src/test/resources/infinispan-core-tests-component-metadata.dat
	counter/src/main/resources/infinispan-clustered-counter-component-metadata.dat
	extended-statistics/src/main/resources/infinispan-extended-statistics-component-metadata.dat
	jcache/embedded/src/main/resources/infinispan-jcache-component-metadata.dat
	lock/src/main/resources/
	multimap/src/main/resources/infinispan-multimap-component-metadata.dat
	query/src/main/resources/infinispan-query-component-metadata.dat
	query/src/test/resources/infinispan-query-tests-component-metadata.dat
	remote-query/remote-query-server/src/main/resources/infinispan-remote-query-server-component-metadata.dat
	server/core/src/main/resources/
	server/hotrod/src/main/resources/infinispan-server-hotrod-component-metadata.dat
	server/integration/infinispan/src/main/resources/infinispan-server-infinispan-component-metadata.dat
	tasks/manager/src/main/resources/
	tasks/scripting/src/main/resources/

nothing added to commit but untracked files present (use "git add" to track)
```

**Comment by anistor** [comment]
Actually this ^^^^ is after me messing it up by trying to backport the just the gitignore stuff. We definitely need a bit more.

**Comment by anistor** [comment]
So on a clean build of jdg-72x, after, running some hot-rod tests, I switch to master and have this: 
```
adrian@mojo ~/work/infinispan $ git status 
On branch master
Your branch is ahead of 'origin/master' by 1 commit.
  (use "git push" to publish your local commits)

Untracked files:
  (use "git add <file>..." to include in what will be committed)

	cli/cli-interpreter/src/main/resources/infinispan-cli-interpreter-component-metadata.dat
	core/src/main/resources/infinispan-core-component-metadata.dat
	core/src/test/resources/infinispan-core-tests-component-metadata.dat
	counter/src/main/resources/infinispan-clustered-counter-component-metadata.dat
	extended-statistics/src/main/resources/infinispan-extended-statistics-component-metadata.dat
	jcache/embedded/src/main/resources/infinispan-jcache-component-metadata.dat
	lock/src/main/resources/
	multimap/src/main/resources/infinispan-multimap-component-metadata.dat
	query/src/main/resources/infinispan-query-component-metadata.dat
	query/src/test/resources/infinispan-query-tests-component-metadata.dat
	remote-query/remote-query-server/src/main/resources/infinispan-remote-query-server-component-metadata.dat
	server/core/src/main/resources/infinispan-server-core-component-metadata.dat
	server/hotrod/src/main/resources/infinispan-server-hotrod-component-metadata.dat
	server/integration/infinispan/src/main/resources/infinispan-server-infinispan-component-metadata.dat
	tasks/manager/src/main/resources/
	tasks/scripting/src/main/resources/

nothing added to commit but untracked files present (use "git add" to track)
```

**Comment by danberindei** [comment]
@anistor I think this also means that every time you switch to master you end up with these files in your sources and you won't be able to run tests from IntelliJ because they will try to use the wrong metadata files. I strongly suggest using separate working directories/IntelliJ projects for community and prod, but I'll also try to backport the metadata move from src to target.

**Pull request closed without merging**